packages/elvis-dropdown-lit/src/e-dropdown.ts

```diff
@@ -0,0 +1,169 @@
+      top: calc(100% + 8px);
+      left: 0;
+      width: calc(var(--width) - calc(2 * 8px));
+      display: flex;
+      flex-direction: column;
+      box-shadow: ${unsafeCSS(getShadow('medium'))};
+      border-radius: 8px;
+      z-index: 1;
+      max-height: calc(var(--option-height) * 5.5);
+      overflow-y: auto;
+      background-color: ${unsafeCSS(getThemeColor('background-overlay-1'))};
+    }
+
+    .options e-option {
+      &:hover {
+        z-index: 2;
+        background-color: ${unsafeCSS(getThemeColor('background-hover-1'))};
+      }
+    }
+  `;
+
+  @property({ type: String })
+  label: string;
+
+  @property({ type: String })
+  placeholder: string;
+
+  @property()
+  isOpen = false;
+
+  @property({ type: String })
+  value = '';
+
+  @provide({ context: dropdownContext })
+  dropdownContext: DropdownContext = {
+    value: this.value,
+    updateValue: (value) => {
+      this.value = value;
+      this._selectedOptionLabel = this.findLabelFromValue(value) ?? '';
+      this.triggerValueChangeEvent(value);
+    },
+  };
+
+  @state()
+  private _selectedOptionLabel = '';
+
+  render() {
+    return html`
+      <label>
+        ${this.label && html`<span>${this.label}</span>`}
+        <div class="input">
+          <input
+            @focus=${this.handleOpen}
+            value=${this._selectedOptionLabel}
+            placeholder=${this.placeholder}
+          />
+          <div class=${classMap({ icon: true, open: this.isOpen })} .innerHTML=${arrowDownIcon}></div>
+        </div>
+      </label>
+      ${this.isOpen
+        ? html`<div class="options">
+            <slot></slot>
+          </div>`
+        : ''}
+    `;
+  }
+
+  connectedCallback(): void {
+    super.connectedCallback();
+    document.addEventListener('click', this.closeOnOutsideClick);
+  }
+
+  disconnectedCallback(): void {
+    super.disconnectedCallback();
+    document.removeEventListener('click', this.closeOnOutsideClick);
+  }
+
+  private handleOpen() {
+    this.isOpen = true;
+  }
+
+  private handleClose = () => {
+    this.isOpen = false;
+  };
+
+  private closeOnOutsideClick = (event: MouseEvent) => {
+    if (!this.contains(event.target as Node)) {
+      this.handleClose();
+    }
+  };
+
+  private findLabelFromValue(value: string) {
+    if (!value) {
+      throw new Error(`No value found, have you set the value on the e-option?`);
+    }
+    const newLabel = Array.from(this.querySelectorAll(`e-option`)).find((option) => option.value === value);
+    if (newLabel) {
+      return newLabel.textContent;
+    }
+    throw new Error(`No option with value ${value} found, have you set the correct value on the e-option?`);
+  }
+
+  private triggerValueChangeEvent(value: string) {
+    this.dispatchEvent(new CustomEvent('valueChange', { detail: { value } }));
+  }
+}
+
+declare global {
+  interface HTMLElementTagNameMap {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
```

Comment:
😆

Reply:
Yeah, Typescript was complaining because this was being injected into the `global interface HTMLElementTagNameMap` both here _and_ in the built dist-files, so this was the easiest quick-fix I could thing of 🥴